docs/styleguide/PyGuide.md

```diff
@@ -719,21 +719,64 @@ long_string = textwrap.dedent('''\
 <a id="37-logging"></a>
 <a id="logging"></a>
 ### 3.7 Logging 
+Use the logger object built into NNCF for all purposes of logging within the NNCF package code.
+Do not use `print(...)` or other ways of output.
+
+Correct:
+
+```python
+from nncf.common.logging import nncf_logger
+
+nncf_logger.info("This is an info-level log message")
+```
+
+Wrong:
+```python
+print("This is an info-level log message")
+```
 
 For logging functions that expect a pattern-string (with %-placeholders) as
-their first argument: Always call them with a string literal (not an f-string!)
-as their first argument with pattern-parameters as subsequent arguments. Some
-logging implementations collect the unexpanded pattern-string as a queryable
-field. It also prevents spending time rendering a message that no logger is
-configured to output.
+their first argument - consider calling them with a string literal (not an f-string!)
+as their first argument with pattern-parameters as subsequent arguments, if constructing the log message takes a long
+time or is otherwise hurtful to performance.
 
 ```python
-# Correct:
 import nncf
-from nncf.common.utils.logger import logger as nncf_logger
+from nncf.common.logging import nncf_logger
```

Comment:
What reason of renaming `from nncf.common.utils.logger import logger as nncf_logger` -> `from nncf.common.logging import nncf_logger`? Honestly I don't see any value in it.

Reply:
It is easier in IDE (PyCharm or other rich IDEs) to type `nncf_logger`, and then Ctrl+Enter to trigger automatic import of the symbol from the correct location since `nncf_logger` is a unique name.